warehouse/oidc/tasks.py

```diff
@@ -0,0 +1,24 @@
+from warehouse import tasks
+from warehouse.metrics import IMetricsService
+from warehouse.packaging.models import Project
+
+
+@tasks.task(ignore_result=True, acks_late=True)
+def compute_oidc_metrics(request):
+    metrics = request.find_service(IMetricsService, context=None)
+
+    projects_using_oidc = (
+        request.db.query(Project).distinct().join(Project.oidc_publishers)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏The query as is is likely to perform sequential scans on the `projects` table which is now in the 400k+ range - we may need to add an index. Have you looked into that already?

Reply:
Also, technically, this is just counting projects that have a publisher configured, but that doesn't mean they're actually _using_ it.

We might want to count `FileEvents` from OIDC publishers instead:

```sql
select count(*) from file_events where additional->>'publisher_url' is not null;
```

and if we want to count # of projects, we can do the necessary joins from that.

Reply:
Per @di's suggestion I've distinguished between projects that have configured OIDC and have successfully published a release with OIDC. Hopefully the join-chain is what you were expecting?

@miketheman Do we need to change either of these queries to optimize them further? My SQLAlchemy-fu is a bit lacking so any suggestions would be helpful :pray:

Reply:
> Do we need to change either of these queries to optimize them further?

I'm gonna say "not yet" for now - let's let the task run and see how it does.